### ISSUE TITLE
Axis takes space required but does not yield it

### ISSUE THREAD

**Issue by Streyder** [pyqtgraph/pyqtgraph#1053]
Hey Guys,

i am currently looking to create a plot that shows sizes from 1Å to several µm. so 1E-10 to 10E-6 maybe. Obviously I chose a logarithmic axis to do so. On this axis, I don't want to change the unit away from nm and I also don't want scientific notation for my numbers. So 1µm should not show up as 1E-6 or other scaled factors but as 1000 on my nanometer scale. And yes I am aware this will lead to number like 100000 on my graph. Thats fine and wanted!

Down below is my solution which works, but has a the issue mentioned below. Although not strictly related to the issue itself: is my approach the right one? Is there an easier, better or more stable way to do this?

I tried confining the y-axis by setting the viewbox limits but that caused all sorts of weird bugs. Viewbox limits and log axes don't seem to work together very well. But that for another issue.

Onto the issue itself:

### Short description
If tick labels become large, the axis taxes more and more space to show them. To do so the axis grows. But once labels get smaller again, the axis does not "shrink" again so the plot does not increase in size again.

To see this: use the example and zoom the y-axis (I locked the x-axis for convenience). As numbers get bigger the axis grows. If you zoom back in and numbers get smaller, the axis does not get smaller. In extreme cases this leads to the graph no longer visible at all. I couldn't find a way to recover from this.

### Code to reproduce
```python
import sys
from PyQt5 import QtWidgets
import pyqtgraph as pg
import numpy as np


class Example(QtWidgets.QMainWindow):
    def __init__(self):
        super(Example, self).__init__()

        self.resize(800, 600)

        glw = pg.GraphicsLayoutWidget(self)
        self.setCentralWidget(glw)

        x_vals = [0, 1, 2, 3,]
        y_vals1 = [1e-10, 6e-9, 10e-9, 3e-6]

        plot1 = pg.PlotItem(axisItems={'left': NanoMeterScaledLogAxis(orientation='left')})
        plot1.setLogMode(False, True)
        plot1.setMouseEnabled(x=False)
        left_axis = plot1.getAxis("left")
        left_axis.setLabel("Radius [nm]")
        left_axis.enableAutoSIPrefix(False)

        red_line = plot1.plot(x=x_vals, y=y_vals1, pen="r")

        glw.addItem(plot1, 0, 0)


class NanoMeterScaledLogAxis(pg.AxisItem):
    def __init__(self, *args, **kwargs):
        super(NanoMeterScaledLogAxis, self).__init__(*args, **kwargs)

    def tickStrings(self, values, scale, spacing):
        return ["{:.2f}".format(np.power(10, value)*1e9) for value in values]


if __name__ == '__main__':
    app = QtWidgets.QApplication(sys.argv)
    ex = Example()
    ex.show()
    sys.exit(app.exec_())
```

### Expected behavior
In all scenarios, independent of the the zoom factor the y-axis should only take the space necessary.

### Real behavior
Y-Axis takes space from the graph, but does not yield it once it is no longer required.

### Tested environment(s)

* PyQtGraph version: Current development branch
* Qt Python binding: PyQt 5.12.3
* Python version: 3.7.4
* NumPy version: 1.16.3
* Operating system: Win10
* Installation method: pip

**Comment by j9ac9k** [comment]
Hi @Streyder ,

I suspect your implementation is probably the best thing to do given your needs.  I actually created #943 because I saw the need for something along the lines you describe.

If you are up for submitting a PR involving this kind of functionality going directly into `AxisItem` we would certainly review it.

Sorry I can't provide more information, kudos on you for subclassing and making pyqtgraph work for you.
